File: src/Library/ObjectModel/IRKeyEventArgs.vb
Public Class IRKeyEventArgs
    Inherits Pixa.Soundbridge.Library.SoundbridgeEventArgs

    Private _command As IRCommand
    Private _isHandled As Boolean = False

    Public Sub New(ByVal soundbridge As Soundbridge, ByVal command As IRCommand)
        MyBase.New(soundbridge)
        _command = command
    End Sub

    Public ReadOnly Property Command() As IRCommand
        Get
            Return _command
        End Get
    End Property

    Public ReadOnly Property IsHandled() As Boolean
        Get
            Return _isHandled
        End Get
    End Property

    Public Sub Handle()
        _isHandled = True
    End Sub
End Class

File: src/Pixa.Soundbridge/IRKeyEventArgs.cs
namespace Pixa.Soundbridge
{
    public class IRKeyEventArgs : SoundbridgeEventArgs
    {
        private IRCommand _command;
        private bool _isHandled = false;

        public IRKeyEventArgs(Soundbridge soundbridge, IRCommand command) : base(soundbridge)
        {
            _command = command;
        }

        public IRCommand Command
        {
            get
            {
                return _command;
            }
        }

        public bool IsHandled
        {
            get
            {
                return _isHandled;
            }
        }

        public void Handle()
        {
            _isHandled = true;
        }
    }
}